src/bootstrap/lib.rs

```diff
@@ -857,6 +857,8 @@ impl Build {
         if let Some(linker) = self.config.target_config.get(&target).and_then(|c| c.linker.as_ref())
         {
             Some(linker)
+        } else if target.contains("vxworks") {
+            Some(self.cxx[&target].path())
```

Comment:
This also seems suspicious -- C++ compiler front-ends are indeed often also valid linker  frontends, but I would expect that to not be something Rust should know; end-users should configure `linker = ` in the vxworks target.

Reply:
Before the change, the linker would be set to C compiler(wr-cc) based on the setting in cc-rs, but for VxWorks the linker needs to be CXX compiler(wr-c++) as the exception functions are in one C++ library.

Reply:
The change here is to fix the problem in building the test suites inside rust. The end-user will not see this problem as I believe for end-user the "linker" would be coming from the settting in librustc_target spec file for one target.

Reply:
It seems like that change should be made upstream in cc-rs perhaps? Presumably this change is necessary for all Rust projects, not just bootstrap?

Reply:
The problem occurs when using bootstrap to get the linker, so I think the change should be in bootstrap.

I have updated cc-rs to use wr-cc for C file and wr-c++ for C++ file, I don't think there is any place else I should change in cc-rs.

Reply:
I think I would prefer if the change was just in cc_detect.rs, and we did something like `linker.insert(target, cxx);` with a comment explaining why (basically the explanation you already gave).

Reply:
Can you give more information about "linker.insert(target, cxx)"? 
It seems there is no ready-to-use method to change the linker for the 'build'.

Reply:
Okay, yeah, taking a closer look that seems more complicated than I had hoped.

Please add comments at both places though explaining why vxworks is special (basically just citing the C++ exception functions, I guess).

Reply:
Just added the comment, please help to review it. Thanks. --Baoshan